aws/data_source_aws_codecommit_repository.go

```diff
@@ -0,0 +1,74 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/codecommit"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceAwsCodeCommitRepository() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsCodeCommitRepositoryRead,
+
+		Schema: map[string]*schema.Schema{
+			"repository_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateMaxLength(100),
+			},
+
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"repository_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"clone_url_http": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"clone_url_ssh": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceAwsCodeCommitRepositoryRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).codecommitconn
+
+	repositoryName := d.Get("repository_name").(string)
+	input := &codecommit.GetRepositoryInput{
+		RepositoryName: aws.String(repositoryName),
+	}
+
+	out, err := conn.GetRepository(input)
+	if err != nil {
+		if isAWSErr(err, codecommit.ErrCodeRepositoryDoesNotExistException, "") {
+			log.Printf("[WARN] CodeCommit Repository (%s) not found, removing from state", d.Id())
+			d.SetId("")
+			return nil
```

Comment:
Currently, data sources should return a direct error instead of silently failing returning no results and likely returning a `resource data.XXX.XXX not found for` error later on. 😄